Corona-Warn-App/src/main/java/de/rki/coronawarnapp/CoronaWarnApplication.kt

```diff
@@ -65,49 +35,15 @@ class CoronaWarnApplication : Application(), HasAndroidInjector {
-    override fun androidInjector(): AndroidInjector<Any> = androidInjector
-
-    @Inject lateinit var watchdogService: WatchdogService
-    @Inject lateinit var taskController: TaskController
-    @Inject lateinit var foregroundState: ForegroundState
     @Inject lateinit var workManager: WorkManager
-    @Inject lateinit var configChangeDetector: ConfigChangeDetector
-    @Inject lateinit var ewRiskLevelChangeDetector: EwRiskLevelChangeDetector
-    @Inject lateinit var combinedRiskLevelChangeDetector: CombinedRiskLevelChangeDetector
-    @Inject lateinit var deadmanNotificationScheduler: DeadmanNotificationScheduler
-    @Inject lateinit var contactDiaryWorkScheduler: ContactDiaryWorkScheduler
-    @Inject lateinit var dataDonationAnalyticsScheduler: DataDonationAnalyticsScheduler
-    @Inject lateinit var notificationHelper: GeneralNotifications
-    @Inject lateinit var deviceTimeHandler: DeviceTimeHandler
-    @Inject lateinit var autoSubmission: AutoSubmission
-    @Inject lateinit var coronaTestRepository: CoronaTestRepository
-    @Inject lateinit var autoCheckOut: AutoCheckOut
-    @Inject lateinit var traceLocationDbCleanupScheduler: TraceLocationDbCleanUpScheduler
-    @Inject lateinit var shareTestResultNotificationService: ShareTestResultNotificationService
-    @Inject lateinit var exposureWindowRiskWorkScheduler: ExposureWindowRiskWorkScheduler
-    @Inject lateinit var presenceTracingRiskWorkScheduler: PresenceTracingRiskWorkScheduler
-    @Inject lateinit var pcrTestResultScheduler: PCRResultScheduler
-    @Inject lateinit var raTestResultScheduler: RAResultScheduler
-    @Inject lateinit var pcrTestResultAvailableNotificationService: PCRTestResultAvailableNotificationService
-    @Inject lateinit var raTestResultAvailableNotificationService: RATTestResultAvailableNotificationService
-    @Inject lateinit var testCertificateRetrievalScheduler: TestCertificateRetrievalScheduler
-    @Inject lateinit var environmentSetup: EnvironmentSetup
-    @Inject lateinit var localStatisticsRetrievalScheduler: LocalStatisticsRetrievalScheduler
     @Inject lateinit var imageLoaderFactory: ImageLoaderFactory
-    @Inject lateinit var dccStateCheckScheduler: DccStateCheckScheduler
-    @Inject lateinit var securityProvider: SecurityProvider
-    @Inject lateinit var recycleBinCleanUpScheduler: RecycleBinCleanUpScheduler
-    @Inject lateinit var vaccinationStorage: VaccinationStorage
-    @Inject lateinit var cclConfigurationUpdaterScheduler: CclConfigurationUpdateScheduler
-    @Inject lateinit var familyTestResultRetrievalScheduler: FamilyTestResultRetrievalScheduler
-    @Inject lateinit var dccValidityStateChangeObserver: DccValidityStateChangeObserver
-    @Inject lateinit var dccRevocationUpdateScheduler: DccRevocationUpdateScheduler
-
-    @AppScope
-    @Inject lateinit var appScope: CoroutineScope
-
+    @Inject lateinit var foregroundState: ForegroundState
+    @Inject lateinit var initializers: Set<@JvmSuppressWildcards Initializer>
```

Comment:
would not `Lazy` be better ? afaik Provider creates new instances every time `get` is called

Reply:
But we call them only once inside this class, so i don't really see the benefit of using Lazy instead of Provider.
Btw. most of them are Singletons anyway, so Dagger ensures to create only 1 instance overall.

Reply:
Ok, I don't think either way will have a big impact even direct injection , I changed it anyway

Reply:
Sure, it won't have a big impact, but it's just a little optimization.